protocol/common/shard.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import "github.com/status-im/status-go/protocol/transport"
+
+type Shard struct {
+	Cluster uint16 `json:"cluster"`
+	Index   uint16 `json:"index"`
+}
+
+func (s *Shard) TransportShard() *transport.Shard {
+	if s == nil {
```

Comment:
This looks suspicious, the caller should make sure that the callee is not `nil` in general

Reply:
I agree that it is somewhat suspicious but it was a deliverate choice because using this approach does have the advantage of returning `nil` immediately instead of having the callee check if the shard is nil. (Since `.TransportShard()` is called in a lot of places, it saves us having to  repeat this in a lot of places, or have to create an function just to do:
```
var pubsubTopic := relay.DefaultPubsubTopic
if community.Shard() != nil {
   pubsubTopic = transport.GetPubsbuTopic(community.Shard().TransportShard())
}
```
While with this approach I can do this directly:
```
pubsubTopic = transport.GetPubsubTopic(community.Shard().TransportShard())
```
and `pubsubTopic` will have either the default pubsub topic or the shard topic